sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementShareConstants.cs

```diff
@@ -22,56 +19,70 @@ internal class SourceCheckpointData
+            internal const int FileChangedOnLengthIndex = FileChangedOnIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveContentTypeIndex = FileChangedOnLengthIndex + OffsetLengthEncodedSize;
+            internal const int ContentTypeOffsetIndex = PreserveContentTypeIndex + PreserveEncodedSize;
+            internal const int ContentTypeLengthIndex = ContentTypeOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveContentEncodingIndex = ContentTypeLengthIndex + OffsetLengthEncodedSize;
+            internal const int ContentEncodingOffsetIndex = PreserveContentEncodingIndex + PreserveEncodedSize;
+            internal const int ContentEncodingLengthIndex = ContentEncodingOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveContentLanguageIndex = ContentEncodingLengthIndex + OffsetLengthEncodedSize;
+            internal const int ContentLanguageOffsetIndex = PreserveContentLanguageIndex + PreserveEncodedSize;
+            internal const int ContentLanguageLengthIndex = ContentLanguageOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveContentDispositionIndex = ContentLanguageLengthIndex + OffsetLengthEncodedSize;
+            internal const int ContentDispositionOffsetIndex = PreserveContentDispositionIndex + PreserveEncodedSize;
+            internal const int ContentDispositionLengthIndex = ContentDispositionOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveCacheControlIndex = ContentDispositionLengthIndex + OffsetLengthEncodedSize;
+            internal const int CacheControlOffsetIndex = PreserveCacheControlIndex + PreserveEncodedSize;
+            internal const int CacheControlLengthIndex = CacheControlOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveFileMetadataIndex = CacheControlLengthIndex + OffsetLengthEncodedSize;
+            internal const int FileMetadataOffsetIndex = PreserveFileMetadataIndex + PreserveEncodedSize;
+            internal const int FileMetadataLengthIndex = FileMetadataOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int PreserveDirectoryMetadataIndex = FileMetadataLengthIndex + OffsetLengthEncodedSize;
+            internal const int DirectoryMetadataOffsetIndex = PreserveDirectoryMetadataIndex + PreserveEncodedSize;
+            internal const int DirectoryMetadataLengthIndex = DirectoryMetadataOffsetIndex + OffsetLengthEncodedSize;
+
+            internal const int VariableLengthStartIndex = DirectoryMetadataLengthIndex + OffsetLengthEncodedSize;
+        }
+
+        internal class ResourceProperties
+        {
+            internal const string FileAttributes = "FileAttributes";
+            internal const string FilePermissionKey = "FilePermissionKey";
+            internal const string FileMetadata = "FileMetadata";
+            internal const string DirectoryMetadata = "DirectoryMetadata";
```

Comment:
I think for the properties (unlike the options) we would only actually need one metadata field? It would always be one or the other depending on the resource type right? This would also allow Blob/File to use the same key for metadata and allow for metadata copying from Blob -> File and File -> Blob (which would not currently work)

Reply:
Would it be better to change "FileMetdata" to "Metadata" so then it can do Blob <-> File transfers? Then leave DirectoryMetdata for just directory metedata?

Reply:
If we lose this conversation due to my changes from moving these constants to `DataMovementConstants` just go those constants.

Reply:
Hmmm, good question. In a File -> Blob copy, what does a Directory become? Do we still create an empty Blob for that or no? If we do, do we want to copy the directory metadata to that?

Reply:
For File->Blob, if the subdirectory is empty, we don't create the virtual blob directory because we would need to create an empty blob directory stub.
But if there's blobs in the virtual directory, then we just create the blob paths and then the virtual directory does get created.
We won't copy any directory property or metadata since there's no actual directory in blobs.
That might just be something we document that if the destination is a different type and does not have a certain property we drop it.

Reply:
Done removed directory metadata